pkg/pipeline/sink/m3u8/reader.go

```diff
@@ -0,0 +1,62 @@
+package m3u8
+
+import (
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Playlist struct {
+	Version        int
+	MediaType      string
+	TargetDuration int
+	Segments       []*Segment
+	Closed         bool
+}
+
+type Segment struct {
+	ProgramDateTime time.Time
+	Duration        float64
+	Filename        string
+}
+
+func ReadPlaylist(filename string) (*Playlist, error) {
+	b, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(string(b), "\n")
+	version, _ := strconv.Atoi(strings.Split(lines[1], ":")[1])
+	mediaType := strings.Split(lines[2], ":")[1]
+	targetDuration, _ := strconv.Atoi(strings.Split(lines[5], ":")[1])
```

Comment:
This will only work with playlists we generated as it assumes the order of the tags. Should we either put a disclaimer at the top of the file, or switch to using regular expression to be more compatible?